dfastmi/kernel/BedLevelCalculator.py

```diff
@@ -0,0 +1,149 @@
+
+    def get_element_wise_maximum(self, dzb: List[numpy.ndarray]) -> numpy.ndarray:
+        """
+        This method gets the element wise maximum from the given bed levels.
+
+        Arguments
+        ---------
+        dzb   : List[numpy.ndarray]
+            List of arrays containing the bed level change at the beginning of each respective discharge period.
+
+        Returns
+        -------
+        dzmax : numpy.ndarray
+            Maximum bed level change.
+        """
+        return numpy.maximum.reduce(dzb)
+
+    def get_element_wise_minimum(self, dzb: List[numpy.ndarray]) -> numpy.ndarray:
+        """
+        This method gets the element wise minimum from the given bed levels.
+
+        Arguments
+        ---------
+        dzb   : List[numpy.ndarray]
+            List of arrays containing the bed level change at the beginning of each respective discharge period.
+
+        Returns
+        -------
+        dzmin : numpy.ndarray
+            minimum bed level change.
+        """
+        return numpy.minimum.reduce(dzb)
+
+    def get_linear_average(self, number_of_days: Vector, dzb: List[numpy.ndarray]) -> numpy.ndarray:
+        """
+        This method gets the linear average from the given bed levels.
+
+        Arguments
+        ---------
+        number_of_days : Vector
+            A vector of periods indicating the number of days during which each discharge applies.
+        dzb   : List[numpy.ndarray]
+            List of arrays containing the bed level change at the beginning of each respective discharge period.
+
+        Returns
+        -------
+        dzgem : numpy.ndarray
+            Yearly mean bed level change.
+        """
+        dzgem = 0
+        for i in range(self.number_of_periods):
+            dzgem += dzb[i] * (number_of_days[i] + number_of_days[i-1]) / 2
+        return dzgem
+
+    def get_bed_level_changes(self, dzq: List[numpy.ndarray], rsigma: Vector) -> List[numpy.ndarray]:
+        """
+        This routine computes the bed level changes.
+        This routine requires that dzq and rsigma have the same length.
+        A stagnant period can be represented by a period with rsigma = 1.
+
+        Arguments
+        ---------
+        dzq : List[numpy.ndarray]
+            A list of arrays containing the equilibrium bed level change for each respective discharge period.
+        rsigma : Vector
+            A tuple of relaxation factors, one for each period.
+
+        Returns
+        -------
+        dzb   : List[numpy.ndarray]
+            List of arrays containing the bed level change at the beginning of each respective discharge period.
+        """
+        vsigma = self.__compute_vsigma(rsigma, dzq)
+        den = self.__compute_denominator(self.number_of_periods, vsigma)
+        dzb = self.__compute_dzb_at_the_beginning_of_each_period(dzq, vsigma, den)
+        return dzb
+
+    def __compute_vsigma(self, rsigma : List[float], dzq: List[numpy.ndarray]) -> List[numpy.ndarray]:
```

Comment:
Should we add descriptions here too? Or is that only needed for the public functions?
Furthermore, the `List[float]` type is equivalent to `Vector` in the other routines.
Yes, the lists are indeed frequently referred to as tuples in the comments. I've used tuples before since that was better for  the static type checker "mypy" and all tuples had exactly length 3. Now, they can have any length and lists are easier for that ...

Reply:
"Private" method do not necessarily need documentation, since they are "private" and no user of the class or module should use it.

I will change the vector and boolvector from a tuple to a list.
I will change the List[float] in this class to vector and update documentation.